src/arrayMethodReduce.js

```diff
@@ -5,7 +5,14 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let value = initialValue === undefined ? this[0] : initialValue;
```

Comment:
it can be done directly in arguments
add a default value to arguments and remove this line